src/rules/noTautologyExpressionRule.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * @license
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as tsutils from "tsutils";
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+
+const TAUTOLOGY_DISCOVERED_ERROR_STRING =
+    "Expression is either a tautology or a contradiction. Binary expression is redundant.";
```

Comment:
> either a tautology or a contradiction

It'd be nice to indicate why the error was detected. How about this phrasing?
```suggestion
    "Both sides of this equality comparison are the same, so the expression is either a tautology or a contradiction.";
```